src/components/Input/Input.tsx

```diff
@@ -1,88 +1,14 @@
-import React, {useEffect, useRef} from 'react';
-import clsx from 'clsx';
-import {Cancel, Search} from '~/icons';
-import './Input.scss';
-import {InputProps, InputSizes, InputVariants} from './Input.types';
+import React from 'react';
+import {InputProps} from './Input.types';
+import {UncontrolledInput} from './UncontrolledInput';
+import {ControlledInput} from './ControlledInput';
 
-export const Input: React.FC<InputProps> = ({
-                                                variant = 'text',
-                                                value = '',
-                                                id,
-                                                placeholder,
-                                                isDisabled = false,
-                                                isReadOnly = false,
-                                                className,
-                                                size = InputSizes.Default,
-                                                icon,
-                                                onClear,
-                                                onChange,
-                                                onBlur,
-                                                onFocus,
-                                                focusOnField = false,
-                                                ...props
-                                            }) => {
-    const classNameProps = clsx(
-        'moonstone-input',
-        {'moonstone-size_big': size === InputSizes.Big},
-        {'moonstone-disabled': isDisabled},
-        className
-    );
-    const inputFilled = value !== '';
-    const inputEmpty = value === '';
-    const searchRef = useRef(null);
+export const Input: React.FC<InputProps> = ({defaultValue, value, ...props}) => {
+    if (typeof value === 'undefined') {
+        return <UncontrolledInput defaultValue={defaultValue} {...props}/>;
```

Comment:
no need to extract defaultValue :
```suggestion
export const Input: React.FC<InputProps> = ({value, ...props}) => {
    if (typeof value === 'undefined') {
        return <UncontrolledInput {...props}/>;
```